docs(k8s): add migration strategies (#2351)
scaleway/docs-content: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx

  By removing or detaching the Public Gateway from the Private Network, a node pool with full isolation can lead to a single point of failure in the cluster, as nodes will no longer be able to reach their control plane.
</Message>

## What are the migration stragies provided?

As we transition towards Private Networks for Kapsule clusters, we understand that you may have varying needs and preferences when it comes to migrating your existing clusters.

> We provide two distinct migration strategies tailored to various requirements and limitations.

Reviewer: ```suggestion
We recommend two distinct migration strategies tailored to various requirements and limitations.
```

| **Implementation:**                                                                                                                                                                                                                                                                                                                                        | **Implementation:**                                                                                                                                                                                                                                                                              |

> | 1. Access migration API: [Scaleway API Documentation](https://www.scaleway.com/en/developers/api/kubernetes/#path-clusters-migrate-an-existing-cluster-to-a-private-network-cluster).<br />2. Initiate migration via Scaleway Console's Private Network tab.<br />3. For Terraform users: Attach a Private Network ID to the cluster resource to activate migration. | 1. Establish a new Kapsule cluster on a Private Network.<br />2. Manually deploy services & containers.<br />3. Run both clusters simultaneously.<br />4. Once the new cluster is verified functional, redirect traffic to it.<br />5. Decommission the old cluster. Configuration intricacy can affect this method. |

Reviewer 2: ```suggestion
| 1. Access migration API: [Scaleway API Documentation](https://www.scaleway.com/en/developers/api/kubernetes/#path-clusters-migrate-an-existing-cluster-to-a-private-network-cluster).<br />2. Initiate migration via Scaleway Console's Private Network tab.<br />3. For Terraform users: Attach a Private Network ID to the cluster resource to activate migration. | 1. Establish a new Kapsule cluster on a Private Network.<br />2. Manually deploy services and containers.<br />3. Run both clusters simultaneously.<br />4. Once the new cluster is verified, redirect traffic to it.<br />5. Decommission the old cluster. Configuration intricacy can affect this method. |
```

| **Pros:**                                                                                                                                                                                                                                                                                                                                                  | **Pros:**                                                                                                                                                                                                                                                                                        |

> |   * Detailed migration guidelines: [Scaleway Documentation](https://www.scaleway.com/en/docs/containers/kubernetes/reference-content/secure-cluster-with-private-network/#how-can-i-migrate-my-existing-clusters-to-private-networks).<br />  * Efficient migration via the provided API.                                                                         |  * Potential for negligible downtime.<br />  * Thorough testing & validation of the new cluster prior to traffic rerouting.                                                                                                                                                                            |

Reviewer 2: ```suggestion
|   * Detailed migration guidelines: [Scaleway Documentation](/containers/kubernetes/reference-content/secure-cluster-with-private-network/#how-can-i-migrate-my-existing-clusters-to-private-networks).<br />  * Efficient migration via the provided API.                                                                         |  * Potential for negligible downtime.<br />  * Thorough testing and validation of the new cluster prior to traffic rerouting.                                                                                                                                                                            |
```

| **Cons:**                                                                                                                                                                                                                                                                                                                                                  | **Cons:**                                                                                                                                                                                                                                                                                        |

> | Downtime during migration.                                                                                                                                                                                                                                                                                                                      | 1. Necessitates a new cluster, leading to extra expenses.<br />2. Complexity varies with workload & configuration.<br />3. Replication and validation can prolong migration.                                                                                                                               |

Reviewer: ```suggestion
| Downtime during migration.                                                                                                                                                                                                                                                                                                                      | 1. Needs a new cluster, leading to extra expenses.<br />2. Complexity varies with workload & configuration.<br />3. Replication and validation can prolong migration.                                                                                                                               |
```

## What future options will there be for isolation?

- **Control plane in isolation** with nodes and API communicating in the same isolated network. The CNI’s network policies will restrict/allow a range of IPs or ports to control who can access the API server.

> - **End of support for legacy public clusters.** All clusters will be required to [migrate to controlled isolation (Private Networks)](#how-can-i-migrate-my-existing-clusters-to-private-networks) by Q1/2024.

Reviewer 2: ```suggestion
- **End of support for legacy public clusters.** All clusters will need to [migrate to controlled isolation (Private Networks)](#how-can-i-migrate-my-existing-clusters-to-private-networks) by Q1/2024.
```
